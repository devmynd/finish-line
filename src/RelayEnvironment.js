import { Component } from 'react'
import PropTypes from 'prop-types'
import { RelayEnvironmentProvider } from './RelayEnvironmentProvider'

export class RelayEnvironment extends Component {
  static propTypes = {
    children: PropTypes.func.isRequired
  }

  static contextTypes = RelayEnvironmentProvider.childContextTypes

  render () {
    return this.props.children(this.context.relayEnvironment)
  }
}
