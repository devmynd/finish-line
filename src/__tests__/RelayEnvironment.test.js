/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { RelayEnvironment } from '../RelayEnvironment'

describe('#render', () => {
  let relayEnvironment
  let children
  let subject

  beforeEach(() => {
    children = jest.fn(() => <div>rendered!</div>)
    relayEnvironment = {
      commitMutation: jest.fn(),
      current: 'something',
      refresh: jest.fn()
    }
    subject = shallow(<RelayEnvironment>
      {children}
    </RelayEnvironment>, { context: { relayEnvironment } })
  })

  it('renders the result of calling the children function with relayEnvironment from context', () => {
    expect(children).toHaveBeenCalledWith(relayEnvironment)
    expect(subject).toMatchSnapshot()
  })
})
