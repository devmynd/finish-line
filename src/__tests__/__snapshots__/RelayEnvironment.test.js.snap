// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render renders the result of calling the children function with relayEnvironment from context 1`] = `
<div>
  rendered!
</div>
`;
